renderdoc/driver/vulkan/vk_initstate.cpp

```diff
@@ -1844,43 +1948,21 @@ void WrappedVulkan::Apply_InitialState(WrappedVkRes *live, const VkInitialConten
     if(IsBlockFormat(fmt))
       bufAlignment = (VkDeviceSize)GetByteSize(1, 1, 1, fmt, 0);
 
-    // first update the live image layout into destination optimal (the initial state
-    // image is always and permanently in source optimal already).
-    dstimBarrier.newLayout = VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL;
-    dstimBarrier.dstAccessMask = VK_ACCESS_TRANSFER_WRITE_BIT;
-
-    VkCommandBuffer extQCmd = VK_NULL_HANDLE;
-
-    if(dstimBarrier.srcQueueFamilyIndex != dstimBarrier.dstQueueFamilyIndex)
-    {
-      extQCmd = GetExtQueueCmd(dstimBarrier.srcQueueFamilyIndex);
-
-      vkr = ObjDisp(extQCmd)->BeginCommandBuffer(Unwrap(extQCmd), &beginInfo);
-      RDCASSERTEQUAL(vkr, VK_SUCCESS);
-    }
-
-    for(size_t si = 0; si < m_ImageLayouts[id].subresourceStates.size(); si++)
-    {
-      dstimBarrier.subresourceRange = m_ImageLayouts[id].subresourceStates[si].subresourceRange;
-      dstimBarrier.oldLayout = m_ImageLayouts[id].subresourceStates[si].newLayout;
-
-      SanitiseOldImageLayout(dstimBarrier.oldLayout);
-
-      dstimBarrier.srcAccessMask = VK_ACCESS_ALL_WRITE_BITS | MakeAccessMask(dstimBarrier.oldLayout);
-
-      DoPipelineBarrier(cmd, 1, &dstimBarrier);
-
-      if(extQCmd != VK_NULL_HANDLE)
-        DoPipelineBarrier(extQCmd, 1, &dstimBarrier);
-    }
-
-    if(extQCmd != VK_NULL_HANDLE)
-    {
-      vkr = ObjDisp(extQCmd)->EndCommandBuffer(Unwrap(extQCmd));
-      RDCASSERTEQUAL(vkr, VK_SUCCESS);
-
-      SubmitAndFlushExtQueue(dstimBarrier.srcQueueFamilyIndex);
-    }
+    std::vector<VkBufferImageCopy> copyRegions;
+    std::vector<VkImageSubresourceRange> clearRegions;
+
+#define INIT_REGION()                                                                           \
+  if(!initialized)                                                                              \
+    copyRegions.push_back(region);                                                              \
```

Comment:
If either side of the if/else needs braces I prefer to have it on both. It's a bit arbitrary but feels easier to parse that they are paired up.

Reply:
Done